Provider/src/test/java/com/spotify/confidence/EventSenderIntegrationTest.kt

```diff
@@ -124,6 +124,68 @@ class EventSenderIntegrationTest {
         }
     }
 
+    @Test
+    fun payload_on_emit() = runTest {
+        val eventStorage = EventStorageImpl(mockContext)
+        val testDispatcher = UnconfinedTestDispatcher(testScheduler)
+        val batchSize = 1
+        val uploadedEvents: MutableList<Event> = mutableListOf()
+        val flushPolicy = object : FlushPolicy {
+            private var size = 0
+            override fun reset() {
+                size = 0
+            }
+
+            override fun hit(event: Event) {
+                size++
+            }
+
+            override fun shouldFlush(): Boolean {
+                return size >= batchSize
+            }
+        }
+        val uploader = object : EventSenderUploader {
+            override suspend fun upload(events: EventBatchRequest): Boolean {
+                uploadedEvents.addAll(events.events)
+                return false
+            }
+        }
+        val engine = EventSenderEngineImpl(
+            eventStorage,
+            clientSecret,
+            flushPolicies = listOf(flushPolicy),
+            dispatcher = testDispatcher,
+            sdkMetadata = SdkMetadata("kotlin_test", ""),
+            uploader = uploader
+        )
+        engine.emit(
+            eventName = "my_event",
+            message = mapOf(
+                "a" to ConfidenceValue.Integer(0),
+                "message" to ConfidenceValue.Integer(1)
+            ),
+            context = mapOf(
+                "a" to ConfidenceValue.Integer(2),
+                "message" to ConfidenceValue.Integer(3)
```

Comment:
I think this is going a bit overboard in this specific test.
Can we move it to a separate one titled something with `handles_message_key_collision`?

Reply:
I renamed the test